src/main/java/htsjdk/samtools/filter/ReadNameFilter.java

```diff
@@ -104,19 +94,8 @@ public boolean filterOut(final SAMRecord record) {
      */
     @Override
     public boolean filterOut(final SAMRecord first, final SAMRecord second) {
-        if (includeReads) {
-            if (readNameFilterSet.contains(first.getReadName()) &&
-                readNameFilterSet.contains(second.getReadName())) {
-                return false;
-            }
-        } else {
-            if (!readNameFilterSet.contains(first.getReadName()) &&
-                !readNameFilterSet.contains(second.getReadName())) {
-                return false;
-            }
-        }
-
-        return true;
+        return (readNameFilterSet.contains(first.getReadName()) &&
+                readNameFilterSet.contains(second.getReadName())) != includeReads;

```

Comment:
Drew out a truth table showing that the new version of this method is not the same as the old version for all cases (the two columns for contains correspond to the two tests for first and second):

```
Original version:

                     includeReads:   T     F
contains:
T     T                                      F     T    
T     F                                      T     T
F     T                                      T     T
F     F                                      T     F

New version:

                     includeReads:   T     F
contains:
T     T                                      F     T   
T     F                                      T     F
F     T                                      T     F
F     F                                      T     F
```

I also think the new version is more confusing / bug prone.